pkg/appfile/helm/schema.go

```diff
@@ -0,0 +1,203 @@
+package helm
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"cuelang.org/go/cue"
+	"cuelang.org/go/cue/format"
+	"cuelang.org/go/encoding/openapi"
+	"cuelang.org/go/encoding/yaml"
+	"github.com/getkin/kin-openapi/openapi3"
+	"github.com/pkg/errors"
+	"helm.sh/helm/v3/pkg/chart/loader"
+	"helm.sh/helm/v3/pkg/getter"
+	"helm.sh/helm/v3/pkg/repo"
+
+	"github.com/oam-dev/kubevela/apis/core.oam.dev/v1alpha2"
+)
+
+var (
+	getters = getter.Providers{
+		getter.Provider{
+			Schemes: []string{"http", "https"},
+			New:     getter.NewHTTPGetter,
+		},
+	}
+)
+
+// GetChartValuesJSONSchema fetched the Chart bundle and get JSON schema of Values
+// file.  If the Chart provides a 'values.json.schema' file, use it directly.
+// Otherwise, try to generate a JSON schema based on the Values file.
+func GetChartValuesJSONSchema(ctx context.Context, h *v1alpha2.Helm) ([]byte, error) {
+	releaseSpec, repoSpec, err := unmarshalHelmSpec(h)
+	if err != nil {
+		return nil, errors.WithMessage(err, "Helm spec is invalid")
+	}
+	chartSpec := releaseSpec.Chart.Spec
+	files, err := loadChartFiles(ctx, repoSpec.URL, chartSpec.Chart, chartSpec.Version)
+	if err != nil {
+		return nil, errors.WithMessage(err, "cannot load Chart files")
+	}
+	var values, jsonSchema *loader.BufferedFile
+	for _, f := range files {
+		if f.Name == "values.yaml" || f.Name == "values.yml" {
+			values = f
+			continue
+		}
+		if f.Name == "values.schema.json" {
+			jsonSchema = f
+			continue
+		}
```

Comment:
from the code style aspect, it's better to use a `switch`. And merge the logic L56~59 into it.